### DISCUSSION TITLE
How to enable HTTPS in Superset when running locally

### DISCUSSION THREAD

**Discussion by remoteeng00** [apache/superset#14384] [Q&A / Help]
In order to integrate Superset with my application properly, I need to enable HTTPS. During development I'm running it with `docker-compose`. I couldn't find any useful information on how to do that. I'd be very glad if you could help me. Thank you

**Comment by remoteeng00** [comment]
I still need someone to help me to figure out this issue

**Reply by jagamts1** [reply]
@remoteeng00 Could please raise same question over [stockoverflow](https://stackoverflow.com/questions/tagged/apache-superset). Let me know once you created and i'll post completed flow. Kindly share which version are you running.

**Reply by remoteeng00** [reply]
@jagamts1 I created the question: https://stackoverflow.com/questions/67376623/how-to-enable-https-in-superset-when-running-locally The version I'm running is v1.1.0. Thank you

**Reply by jagamts1** [reply]
@remoteeng00 Answered for your questions on stock-overflow. Let me know if you are facing any issue.

**Reply by remoteeng00** [reply]
@jagamts1 Thank you. I followed the steps, however, I'm having one issue, I explained it in the Stack Overflow comment, please check.

**Reply by remoteeng00** [reply]
Hi @jagamts1 I still have one issue. I posted in the Stack Overflow comment. Did you have a chance to check it?
Please let me know. Looking to hear from you soon. Thank you

**Reply by jagamts1** [reply]
@remoteeng00 I'll look into that now and let's resolve today itself.

**Reply by jagamts1** [reply]
@remoteeng00 I have answered your question. If you are still facing the issue. Let me know.

**Reply by remoteeng00** [reply]
@jagamts1 I followed your instruction in stackoverflow but still have issues. I wrote there in the ticket on stackoverflow. Please check and let me know. Thank you

**Reply by remoteeng00** [reply]
@jagamts1 did you have a chance to check my question in the stackoverflow?
Looking to hear from you.
Thank you

**Reply by jagamts1** [reply]
@remoteeng00 I tested locally and could reproduce the issue in the following manner.
1.  **VIRTUAL_HOST** didn't exist
2.  .certs files don't contain certificates.

Looks like you are missing something. Kindly check the following GIF

Here GIF for the complete flow for adding **HTTPS** on superset docker(firefox). 
[superset_with_https.zip](https://github.com/apache/superset/files/6468008/superset_with_https.zip)

Here GIF to reproducing your issue and resolved it. 
[reproduce_superset_https_issue.zip](https://github.com/apache/superset/files/6468019/reproduce_superset_https_issue.zip)

I hope this will resolve your issue finally.

**Reply by remoteeng00** [reply]
@jagamts1 I'm still having issues. I couldn't get it to work on Firefox, however, got it to work on Firefox Developer Edition. Now there's no Firefox error. However, Superset responds with 502 Bad Gateway error. I think it has to do something with nginx proxy. To me it seems as if `VIRTUAL_HOST: superset.localhost` wasn't there but it actually is there. I attached screenshots of terminal errors below.
https://ibb.co/XJXSXcW
https://ibb.co/GHgSyQd

Do you know what could be wrong? Please let me know. Thank you.

**Reply by apozas00** [reply]
Hi @jagamts1, I have some issue adapting the the steps you posted in stack overflow for docker-compose-image-tag.yml but I couldn't do anything useful. Do you know how can I do in my situation?
